### PR TITLE
Fix ActionGoalStateDemo plan

### PR DESCRIPTION
**Pull request by kmdalal**
### Summary

This PR fixes, by way of a revamp, the `ActionGoalStatusDemo` plan, which is currently the only plan using the `ActionGoalStatus` lookup.   It had two bugs:

1. Several ROS actions (e.g. Pan, Tilt, Pan/Tilt) were not having their goal status topic subscribed to.  

2. Both panning and arm movement would often not resume after a relevant fault was cleared.

_This PR fixes only bug 1.  Bug 2 is still unsolved and the subject of a future PR.  This PR avoids bug 2 by not having the plan attempt to be fault-tolerant like the original version.  It now has sole purpose: testing the mentioned lookup._

Extra changes:

- In the process of fixing this bug, there was an opportunity to streamline the functions by having the goal status topic name derived from the action name instead of passed in.

- The goal status enumeration was moved up into the common plan interface, and an additional 'inactive' status was added.

- The plan can be terminated with a power fault, instead of requiring Control-C to kill the launch.

Caveats:

- The plan is not fault tolerant: this needs additional work, for a future PR.

### Reviewer Tips

It would be easier to look at `ActionGoalStateDemo.plp` in whole rather than the diff, because the plan was greatly changed.

### Test

1. [Release 12 test 10.10.](https://babelfish.arc.nasa.gov/confluence/pages/viewpage.action?pageId=220561639)

2. Verify the OWLAT build:
```
rm -rf <ow_workspace>/build/ow_plexil
catkin build ow_plexil -DOWLAT=ON
```
OWLAT needs no test because no plans use the `ActionGoalState` lookup.

No other release tests should be impacted or need re-testing, because none of them use the `ActionGoalState` lookup.

### CODE REVIEW

**Comment by AstroStucky** [comment]
Both tests pass

File: ow_plexil/src/plans/ActionGoalStateDemo.plp [hunk 1]
```diff
@@ -2,57 +2,78 @@
...
+      else if (Lookup(ActionGoalStatus("Pan")) == ACTION_ABORTED) {
+        log_info ("Pan is aborted.");
+      }
+      else {
+        log_warning ("Either Pan or Tilt are in an unexpected state:");
+        log_warning ("Pan is in state ",
+                     Lookup(ActionGoalStatus("Pan")));
+        log_warning ("Tilt is in state ",
+                     Lookup(ActionGoalStatus("Tilt")));
       }
```
Comment (AstroStucky) [inline ow_plexil/src/plans/ActionGoalStateDemo.plp:71]: It'd be cool if these statement could be triggered only upon state transition instead of performing checks at a regular rate, but I'm not sure how.

**Review by AstroStucky — APPROVED** [review]
The code looks good!

File: ow_plexil/src/plans/ActionGoalStateDemo.plp (hunk 1 above)
Comment (kmdalal) [reply ow_plexil/src/plans/ActionGoalStateDemo.plp:71]: Polling was actually the intention here -- to see what the state is at a steady interval.   Reacting just to changes could be done in the usual way: a start condition that checks if the state is different from a cached state.
